### PR TITLE
added backend profiling and fixed benchmark's install dependency scripts

### LINKED ISSUE #400: Benchmark profiling for backend not working on staging_0_1_1
Followed instructions in https://github.com/pytorch/serve/blob/staging_0_1_1/benchmarks/README.md

In the profiling-> backend section, it talks about setting constant BENCHMARK to true. But this doesn’t seem to exist in the file ts/model_service_worker.py
 
Just adding this const on top of DEBUG into the file and running torchserve and benchmark.py didn’t generate /tmp/tsPythonProfile. Frontend profiling works fine via JProfiler

### PR DESCRIPTION
**Pull request by harshbafna**
## Description
 This PR addresses following :

 - added code changes to enable backend profiling
 - makes benchmark install dependency scripts independent of jmeter version.

Fixes #400

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

 - create a docker image with BENCHMARK parameter set to true in model_service_worker.py

```
git clone https://github.com/pytorch/serve.git
cd serve
git checkout issue_400
cd docker
git clone https://github.com/pytorch/serve.git
cd serve
git checkout issue_400
## set BENCHMARK flag to true
vim ts/model_service_worker.py
cd ..
DOCKER_BUILDKIT=1 docker build --file Dockerfile_dev.cpu -t torchserve:dev .
```
 - run benchmark's install_dependencies.sh

```
cd benchmark
./install_dependencies.sh
```

 - start docker with /tmp directory mapped to local /tmp

```
docker run --rm -it -p 8080:8080 -p 8081:8081 -v /tmp:/tmp torchserve:dev
```

 - start default throughput benchmark

```
cd benchmark
python benchmark.py throughput --ts http://127.0.0.1:8080
```

**Profiling **
Attached generated report for 10 loops in zip format :

[profile.zip](https://github.com/pytorch/serve/files/4681242/profile.zip)

To access report on browser run following
```
unzip profile.zip
pip install snakeviz
snakeviz tsPythonProfile.prof
```

Profiling report generated on Mac : 
[profile_mac.zip](https://github.com/pytorch/serve/files/4681877/profile_mac.zip)

## Checklist:

- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: benchmarks/README.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ Once you have stopped recording, you should be able to analyze the data.  One us
...
-2. Run the benchmark and TorchServe.  They can either be done automatically inside the docker container or separately with the "--ts" flag.
+2. Run the benchmark and TorchServe.  They can either be done automatically inside the docker container or separately with the "--ts" flag. If you are running `TorchServe` inside docker container make sure you map the `/tmp` directory to some local directory on your machine while starting docker.
 3. Run TorchServe directly through gradle (do not use docker).  This can be done either on your machine or on a remote machine accessible through SSH.
 4. Run the Benchmark script targeting your running TorchServe instance.  It might run something like `./benchmark.py throughput --ts https://127.0.0.1:8443`.  It can be run on either your local machine or a remote machine (if you are running remote), but we recommend running the benchmark on the same machine as the model server to avoid confounding network latencies.
 5. Run `snakeviz /tmp/tsPythonProfile.prof` to view the profiling data.  It should start up a web server on your machine and automatically open the page.
```
Comment (chauhang) [inline benchmarks/README.md:156]: Also add steps in readme for installing snakeviz using `pip install snakeviz`

**Comment by chauhang** [comment]
@harshbafna Please update the readme with few missing items highlighted and then merge the PR. I have tested the code changes they look good.

**Comment by maaquib** [comment]
@harshbafna There is overlap with #412 
Please pick up those changes as well and I'll close it out.

**Comment by harshbafna** [comment]
> @harshbafna There is overlap with #412
> Please pick up those changes as well and I'll close it out.

Done.

**Comment by harshbafna** [comment]
> @harshbafna Please update the readme with few missing items highlighted and then merge the PR. I have tested the code changes they look good.

Done.

**Comment by dhanainme** [comment]
The readme example requested have been added. Merging this to do the final benchmark run for v_0_1_1

**Pull request merged**